compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java

```diff
@@ -555,7 +556,14 @@ private BType getMatchingErrorBoundType(BErrorType expType, SymbolEnv env, HashS
         }
         BType reasonType = getMatchingBoundType(expType.reasonType, env, resolvedTypes);
         BType detailType = getMatchingBoundType(expType.detailType, env, resolvedTypes);
-        return new BErrorType(null, reasonType, detailType);
+        BErrorTypeSymbol typeSymbol = new BErrorTypeSymbol(SymTag.ERROR,
```

Comment:
copying symboltable.error symbol's seems not right, it may print the wrong error type. But printing its structure has its own problems. So let's keep this. Ideally, we should fix the structure problem. 

Other option is check 
```
if( reasonType == symTable.string && types.isSame(detailType, symTable.detailType) {
  return symTable.errorType;
}
```